### ISSUE TITLE
_player.stop not working.

### ISSUE THREAD

**Issue by pasrot943** [ryanheise/just_audio#668]
**Which API doesn't behave as documented, and how does it misbehave?**
Just Audio / Just Audio Background, Cannot stop the player with player.stop or with await player.stop. Used to work like a charm.

**Minimal reproduction project**

https://github.com/pasrot943/just_audio/tree/master/just_audio_background/example/lib

**To Reproduce (i.e. user steps, not code)**
1. Tap one of Playlist 1 or 2
2. Press any of the 3 episodes listed.
3. Press Stop or await stop
4. The player stops but not the music in the background.

**Error messages**

```
No error message at all
```

**Expected behavior**
Expected to stop the background music.

**Screenshots**
No need.

**Desktop (please complete the following information):**
 - Win 10

**Smartphone (please complete the following information):**
Android Emulator Nexus 4
Android 11 API 30
**Flutter SDK version**
Flutter 2.10.2
```
I´m using Android Studio (version 4.2)

Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 2.10.2, on Microsoft Windows [Version 10.0.19044.1526], locale sv-SE)
[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[√] Chrome - develop for the web
[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components
[√] Android Studio (version 4.2)
[!] Android Studio
    X android-studio-dir = C:\Program Files\Android\Android Studio2           
    X Android Studio not found at C:\Program Files\Android\Android Studio2
[√] VS Code (version 1.57.1)
[√] Connected device (4 available)
[√] HTTP Host Availability

! Doctor found issues in 2 categories.
```

**Additional context**
No

**Comment by ryanheise** [comment]
Thanks for reporting this. I will take a look, but before then, can you just clarify the exact behaviour you're seeing? Do you mean that `await` never returns, or do you mean that `stop` doesn't actually do anything?

**Comment by pasrot943** [comment]
_player.stop doesn’t stop the player from playing the current audio source.
The Slider stops. The play/pause button changes like it should, but the
audio continues in the background.

My best guess is that it has to do with gradle or kotlin which isn’t my
strongest sides yet.
I’ve tried a lot and changed the versions many times since this error
occurred.

I will add those files in the repo very soon.






onsdag 2 mars 2022 skrev ryanheise ***@***.***>:

**Comment by ryanheise** [comment]
Have you isolated which version introduced the bug?

**Comment by pasrot943** [comment]
No, It took i while until I realised that it didn´t work anymore.

**Comment by Groseuros** [comment]
Hello !

First, @ryanheise thanks a lot for your awesome work here.

I'm also facing this issue on both iOS/Android. I did some tests and it looks like it was introduced in v0.9.19 (working fine in v0.9.18)

Regards

**Comment by ryanheise** [comment]
Perhaps this #672 related?

**Comment by ryanheise** [comment]
I was about to take a look at this, but your minimal reproduction project does not seem "minimal". There is too much in there that is different from the official example that it's hard to see whether the issue is caused by this plugin or by your own code. The correct way is to create a fork of this plugin and modify the official example to reproduce the bug - then I can look at the diff to see what the minimal changes were.

In particular, to understand why the official example works and why some other use case doesn't work.

@Groseuros are you sure yours is exactly the same issue as this? What is it about your use case that is different from the official (working) example? If it's different again, you may need to open a separate issue for it, unless #672 is the same as yours.

**Comment by pasrot943** [comment]
I have been testing with a lot of version v0.9.18 and lower same with Just
Audio Background down to beta 1.
The problem persists unfortunately.
It seems like it's a growing problem since more people are reporting it.

very sad, I was just about to launch :(

Den fre 4 mars 2022 kl 19:13 skrev Robin J ***@***.***>:

**Comment by pasrot943** [comment]
Hello Ryan,

I understand, but my code should not block the stop function because the
same code worked prior to the bug.
BUT the kotlin version and the build.gradle files should be the ones
subjected to investigation in my view.
I lack that knowledge ,I've tried, believe me.
I got Covid right now , but soon I will try to update the minimal
reproduction project.

Thanx for your help.

Den lör 5 mars 2022 kl 07:33 skrev ryanheise ***@***.***>:

**Comment by ryanheise** [comment]
You didn't actually answer my last comment so I can't continue the investigation. When you are ready for me to investigate, please look at my previous comment and address it. I am currently working on fixing the other stop bug I linked to since it is ready for me to investigate.

**Comment by ryanheise** [comment]
I just fixed #672 so if you aren't able to do what I asked in my comment above, you could try out the fix to #672 in the hopes that it might also fix your issue.

**Comment by pasrot943** [comment]
#672 is working! Thank you very much Ryan! You can close this issue! #672 is the answer!  You saved my day!
